x-pack/plugins/fleet/common/services/agent_status.ts

```diff
@@ -57,6 +57,13 @@ export function buildKueryForInactiveAgents() {
   return 'status:inactive';
 }
 
+export function buildKueryForFilteringOutOfflineAgentlessAgents(kuery: string) {
+  if (kuery.length > 0) {
+    return `not (local_metadata.host.hostname : "agentless*" and status:offline) and (${kuery})`;
+  }
+  return `not (local_metadata.host.hostname : "agentless*" and status:offline)`;
+}
+
```

Comment:
Since the base query is repeating you can minimize risk of small mistakes which can be hard to detect in a long string by consolidating it:

```ts
export function buildKueryForFilteringOutOfflineAgentlessAgents(kuery: string) {
  const baseKuery = `not (local_metadata.host.hostname : "agentless*" and status:offline)`;
  return !!kuery.length ? `${baseKuery} and (${kuery})` : baseKuery;
}
```